### ISSUE TITLE
If I wanted to train an RL bot on TripleA, how would I do it?

### ISSUE THREAD

**Issue by cosmojg** [triplea-game/triplea#10217]
Hello! I'm interested in training a reinforcement learning agent to play TripleA, but I'm not sure about the best way to go about it. Currently, my team and I are using a graphical convolutional neural network where the graph is the game board. We've been manually recording games played on a real, physical Axis & Allies game board, as well as training on random games generated with a simulator we built. Now, we're interested in transitioning to TripleA.

**Questions about training:**
- What's the best way to get information about game state?
- Is it possible to record game states?
- Could we feasibly generate a training dataset by running and recording games between the existing bots?
- Could we feasibly generate a training dataset by running and recording games between humans and bots?

**Questions about playing:**
- If we want to play with our RL agent, what's the best way to hook it up to the game?
- Is it possible to programmatically access game state during a live game?
- Is it possible to programmatically input unit placements and movements during a live game?

**General questions:**
- Is there an existing API which could facilitate all of the above?
- Has anyone attempted something similar in the past?
